### PR TITLE
chore(firestore-bundle-server): increased testing

### PR DESCRIPTION
**Pull request by dackers86**
Updated the Firestore Bundle Server extension test coverage.

- [x] Updated dependency versions
- [x] Converted to Jest testing
- [x] Added emulator configuration
- [x] Introduced e2e function testing

### CODE REVIEW

File: _emulator/firestore-debug.log [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Sept 16, 2022 9:09:03 AM com.google.cloud.datastore.emulator.firestore.websocket.WebSocketServer start
```
Comment (Ehesp) [inline _emulator/firestore-debug.log]: These need adding to gitignore

File: _emulator/firestore.rules [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      // This rule allows anyone with your database reference to view, edit,
+      // and delete all data in your database. It is useful for getting
+      // started, but it is configured to expire after 30 days because it
+      // leaves your app open to attackers. At that time, all client
+      // requests to your database will be denied.
+      //
+      // Make sure to write security rules for your app before that time, or
+      // else all client requests to your database will be denied until you
+      // update your rules.
+      allow read, write: if request.time < timestamp.date(2022, 7, 2);
```
Comment (Ehesp) [inline _emulator/firestore.rules]: Should this just be open for testing?

File: firestore-bundle-server/functions/__tests__/functions.test.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+  it("successfully returns a bundle using clientCache", async () => {
+    const bundleName = "with-client-cache";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    console.log(documentMetadata);
```
Comment (Ehesp) [inline firestore-bundle-server/functions/__tests__/functions.test.ts]: Got a number of console.logs in place still

**Review by Ehesp — CHANGES_REQUESTED** [review]

File: firestore-bundle-server/functions/__tests__/functions.test.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using a document", async () => {
```
Comment (Ehesp) [inline firestore-bundle-server/functions/__tests__/functions.test.ts:127]: `xit` on a few of these?

File: _emulator/firestore-debug.log (hunk 1 above)
Comment (dackers86) [reply _emulator/firestore-debug.log]: Updated.

File: firestore-bundle-server/functions/__tests__/functions.test.ts (hunk 4 above)
Comment (dackers86) [reply firestore-bundle-server/functions/__tests__/functions.test.ts:127]: I've added these as placeholders for now. Some errors timeouts appear on a couple the tests, so I wanted to highlight  these before continuing developing them.

File: firestore-bundle-server/functions/__tests__/functions.test.ts (hunk 3 above)
Comment (dackers86) [reply firestore-bundle-server/functions/__tests__/functions.test.ts]: Updated.

File: _emulator/firestore.rules (hunk 2 above)
Comment (dackers86) [reply _emulator/firestore.rules]: Updated.

**Review by Ehesp — APPROVED** [review]

**Pull request merged**